src/v/security/acl.h

```diff
@@ -441,6 +521,58 @@ class resource_pattern_filter {
           std::nullopt, std::nullopt, std::nullopt);
         return filter;
     }
+    friend void read_nested(
+      iobuf_parser& in,
+      resource_pattern_filter& filter,
+      size_t const bytes_left_limit) {
+        using serde::read_nested;
+        auto resource = read_nested<std::optional<security::resource_type>>(
```

Comment:
I was wondering this, but then also wondered if the passed in `filter` shouldn't be modified at all in case deserialization failed, to provide some exception safety. I guess `read_nested` nearly always works on a default constructed entity, and failures unwind past it.